server.go

```diff
@@ -78,3 +67,20 @@ func (s *forwarderServer) Send(ctx context.Context, d *pb.Data) (*pb.Receipt, er
 	// Respond to the start request that the work was accepted.
 	return &pb.Receipt{}, nil
 }
+
+func jsonData(d *pb.Data) []byte {
+	// Create a data message to send back to the dispatcher.
+	data := httpMessage{
+		ResponseTo: d.GetMessageId(),
+		Metadata:   d.GetMetadata(),
+		Content:    d.GetContent(),
+		Directive:  d.GetDirective(),
+	}
+
+	dataJson, error := json.Marshal(data)
+	if error != nil {
+		log.Fatal(error)
+	}
```

Comment:
I know this is somewhat unlikely to happen and that this is just the error handling that was here before moved into a function, but it doesn't feel right. If the data cannot be encoded as json, we'd log an error message and send a post with an empty body to the server? Shouldn't it report the error back to the cloud and bail out?

Reply:
`Fatal` method logs and exits in the same command: https://pkg.go.dev/log#Fatal

Reply:
That's a bit surprising. Anyway, the worker dies and most likely gets restarted, but is the cloud notified somehow?

Reply:
It depends on yggdrasil's error handling for the `Send` method. The method call would be aborted, so it will result in an error on the dispatcher side.

Reply:
Soooo, what is my take away here?